src/search.cpp

```diff
@@ -387,7 +391,33 @@ void MainThread::search() {
       for (Thread* th : Threads)
           if (   th->completedDepth > bestThread->completedDepth
               && th->rootMoves[0].score > bestThread->rootMoves[0].score)
-              bestThread = th;
+              longestPVThread = bestThread = th;
+
+      if (bestThread->rootMoves[0].pv.size() < minPlies) {
+          for (Thread* th : Threads) {
+              // Look for the best thread that meets the minimum move criteria.
+              // Don't take a thread unless it's within the appropriate range of score
+              // eval. 
+              if (th->rootMoves[0].pv.size() <= longestPVThread->rootMoves[0].pv.size()) {
+                  continue;
+              }
+              bool possiblePVReplacement = true;
+              for (unsigned int i = 0; i < longestPVThread->rootMoves[0].pv.size(); ++i) {
+                  if (th->rootMoves[0].pv[i] != longestPVThread->rootMoves[0].pv[i]) {
+                      possiblePVReplacement = false;
+                      break;
+                  }
+              }
+              if (!possiblePVReplacement) {
+                  continue;
+              }
+              if (
+                     ((th->rootMoves[0].pv.size() >= minPlies)
+                  ||  (abs(th->rootMoves[0].score - longestPVThread->rootMoves[0].score) > maxScoreDiff))
```

Comment:
Thanks for this. I believe you are correct, in this case it should have been a less-than - but I also believe that the entire if statement is not doing what I had intended so I'm going to see if I can write it in a more clear manner and ensure it's doing what I wanted it to.